### PR TITLE
Added test for genesis hash

### PR DESCRIPTION
**Pull request by mm-near**
Added a test to verify that genesis hash doesn't change.

### CODE REVIEW

File: nearcore/src/runtime/mod.rs [hunk 1]
```diff
@@ -3199,4 +3201,27 @@ mod test {
         let view_state_value = view_state.get(&key).unwrap().unwrap();
         assert_eq!(state_value, view_state_value);
     }
+
+    /// Check that mainnet genesis hash still matches.
```
Comment (matklad) [inline nearcore/src/runtime/mod.rs:3205]: Maybe expand this a bit? This assumes that the reader understands *why* we want that, might make sense to spent a sentence explaining that.

File: nearcore/src/runtime/mod.rs [hunk 2]
```diff
@@ -3199,4 +3201,27 @@ mod test {
...
+    #[test]
+    fn test_genesis_hash() {
+        let genesis = Genesis::from_file("res/mainnet_genesis.json", GenesisValidationMode::Full);
+        let chain_genesis = ChainGenesis::new(&genesis);
+        let (tempdir, opener) = near_store::NodeStorage::test_opener();
+        let store = opener.open().unwrap();
```
Comment (matklad) [inline nearcore/src/runtime/mod.rs:3211]: Can we use in-memory store? Feels like this test shouldn't write to disk.

File: nearcore/src/runtime/mod.rs [hunk 3]
```diff
@@ -3199,4 +3201,27 @@ mod test {
...
+            store.get_store(Temperature::Hot),
+            &genesis,
+            TrackedConfig::new_empty(),
+            RuntimeConfigStore::new(None),
+        ));
+
+        let block = Chain::make_genesis_block(&*runtime, &chain_genesis).unwrap();
+        assert_eq!(
+            block.header().hash(),
+            &CryptoHash::from_str("EPnLgE7iEq9s7yTkos96M3cWymH5avBAPm3qx3NXqR8H").unwrap()
```
Comment (matklad) [inline nearcore/src/runtime/mod.rs:3224]:
```suggestion
            &"EPnLgE7iEq9s7yTkos96M3cWymH5avBAPm3qx3NXqR8H".parse::<CryptoHash>().unwrap()
```


more idiomatic form, and avoids the import (which I think will trip unused warning in the current code on CI)

**Review by robin-near — APPROVED** [review]

File: nearcore/res/mainnet_genesis.json [hunk 4]
```diff
@@ -0,0 +1,823 @@
+{
```
Comment (mina86) [inline nearcore/res/mainnet_genesis.json:1]: `utils/mainnet-res/res/mainnet_genesis.json` is what you’re looking for.  Available through `near_mainnet_res::mainnet_genesis` function.

File: nearcore/src/runtime/mod.rs (hunk 2 above)
Comment (mina86) [reply nearcore/src/runtime/mod.rs:3211]: You don’t even need full NodeStorage. `near_store::test_utils::create_test_store` will give you Store you can pass to runtime constructor.

File: nearcore/src/runtime/mod.rs (hunk 3 above)
Comment (mina86) [reply nearcore/src/runtime/mod.rs:3224]: Alternatively, `block.header().hash().to_string()` and compare with encoded hash.

**Review by mina86 — APPROVED** [review]
LGTM except for duplication of the resource file.

File: nearcore/src/runtime/mod.rs (hunk 2 above)
Comment (mm-near) [reply nearcore/src/runtime/mod.rs]: thanks - done.

File: nearcore/src/runtime/mod.rs (hunk 3 above)
Comment (mm-near) [reply nearcore/src/runtime/mod.rs]: done.

**Pull request merged**